src/main/resources/static/data-upload.js

```diff
@@ -187,22 +187,19 @@ function handle_session_retrieval() {
         type: "get",
         data:{sessionId:sid},
         success: function (data) {
+            var page;
             //direct user to the selected session
-            if(session.step == "MAPPING") {
-                window.location.href = baseApp+"/views/mapping";
-            }
-            else if(session.step == "PATIENTS") {
-                window.location.href = baseApp+"/views/subjects";
-            }
-            else if(session.step == "EVENTS") {
-                window.location.href = baseApp+"/views/events";
-            }
-            else if(session.step = "OVERVIEW") {
-                window.location.href = baseApp+"/views/overview";
-            }
-            else {
-                console.log('Session step is not recognized.');
+            switch (session.step) {
+                case "MAPPING": page = "mapping"; break;
+                case "FEEDBACK_DATA": page = "feedback-data"; break;
+                case "SUBJECTS": page = "subjects"; break;
+                case "FEEDBACK_SUBJECTS": page = "feedback-subjects"; break;
+                case "EVENTS": page = "events"; break;
+                case "FEEDBACK_EVENTS": page = "feedback-events"; break;
+                case "OVERVIEW": page = "final"; break;
+                default: page = "mapping";
             }
+            window.location.href = baseApp+"/views/"+page;
```

Comment:
Alt+CMD+L lets you format your code automatically and avoid lines like this one (no spaces between an operator and a variable